cloudinit/kindadapter.go

```diff
@@ -27,37 +27,40 @@ import (
 	"sigs.k8s.io/kind/pkg/exec"
 )
 
-var supportedCloudCongfigActions = map[string]cloudCongfigActionBuilder{
-	"write_files": newWriteFilesAction,
-	"runcmd":      newRunCmdAction,
-}
+const (
+	// Supported cloud config modules
+	writefiles = "write_files"
+	runcmd     = "runcmd"
+)
 
-type cloudCongfigActionBuilder func() cloudCongfigAction
+type actionFactory struct{}
+
+func (a *actionFactory) action(name string) action {
+	switch name {
+	case writefiles:
+		return newWriteFilesAction()
+	case runcmd:
+		return newRunCmdAction()
+	default:
+		// TODO Add a logger during the refactor and log this unknown module
+		return newUnknown(name)
```

Comment:
@detiber What do you think of this? We don't have a logger right now but I'm opening an issue to refactor this module a fair bit and get rid of this dynamic generation. That way we can have a clean Run that will 1) write files, 2) run commands easily (or whatever order cloud-init will do things in)